packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -3680,4 +3680,98 @@ describe('TransactionController', () => {
       );
     });
   });
+
+  describe('updateCustodialTransaction', () => {
+    const transactionId = '1';
+    const statusMock = TransactionStatus.unapproved as const;
+    const baseTransaction = {
+      id: transactionId,
+      chainId: toHex(5),
+      status: statusMock,
+      time: 123456789,
+      txParams: {
+        from: ACCOUNT_MOCK,
+        to: ACCOUNT_2_MOCK,
+      },
+    };
+    const transactionMeta: TransactionMeta = {
+      ...baseTransaction,
+      custodyId: '123',
+      history: [{ ...baseTransaction }],
+    };
+    it.each([
+      {
+        newStatus: TransactionStatus.signed as const,
+      },
+      {
+        newStatus: TransactionStatus.submitted as const,
+      },
+      {
+        newStatus: TransactionStatus.failed as const,
+        errorMessage: 'Error mock',
+      },
+    ])(
+      'updates transaction status to $newStatus',
```

Comment:
Shall we assert updated properties to extend this test with given new statuses? (`submittedTime`, `error`, `custodyStatus` etc)

Reply:
I've included the `hash` and `custodyStatus`, however, for the properties `submittedTime` and  `error` I would need to include conditions to the test. I could create an extra test to assert those properties.